From 8358d822e69e41443dcebe67f2a93b36881f305a Mon Sep 17 00:00:00 2001
From: Ben Peled <bpeled@marvell.com>
Date: Sun, 28 Apr 2019 13:15:54 +0300
Subject: [PATCH 181/386] dt-bindings: marvell: replace ap806 with ap80x

Replace ap806 with ap80x which describe both ap807 and ap807
armada clocks.

Change-Id: I6a4af0c3179d1d5f58b818c9ab12f956f7c6c221
Signed-off-by: Ben Peled <bpeled@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7824
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 ...roller.txt => ap80x-system-controller.txt} | 25 ++++++++++++++++---
 1 file changed, 22 insertions(+), 3 deletions(-)
 rename Documentation/devicetree/bindings/arm/marvell/{ap806-system-controller.txt => ap80x-system-controller.txt} (84%)

diff --git a/Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt b/Documentation/devicetree/bindings/arm/marvell/ap80x-system-controller.txt
similarity index 84%
rename from Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt
rename to Documentation/devicetree/bindings/arm/marvell/ap80x-system-controller.txt
index 8f281816a6b8..42876a1a88fb 100644
--- a/Documentation/devicetree/bindings/arm/marvell/ap806-system-controller.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/ap80x-system-controller.txt
@@ -1,7 +1,7 @@
-Marvell Armada AP806 System Controller
+Marvell Armada AP80x System Controller
 ======================================
 
-The AP806 is one of the two core HW blocks of the Marvell Armada 7K/8K
+The AP806/AP807 is one of the two core HW blocks of the Marvell Armada 7K,8K/931x,3900
 SoCs. It contains system controllers, which provide several registers
 giving access to numerous features: clocks, pin-muxing and many other
 SoC configuration items. This DT binding allows to describe these
@@ -9,7 +9,7 @@ system controllers.
 
 For the top level node:
  - compatible: must be: "syscon", "simple-mfd";
- - reg: register area of the AP806 system controller
+ - reg: register area of the AP80x system controller
 
 SYSTEM CONTROLLER 0
 ===================
@@ -158,3 +158,22 @@ ap_syscon1: system-controller@6f8000 {
 		#clock-cells = <1>;
 	};
 };
+
+Device Tree Clock bindings for cluster clock of AP807 Marvell. Each
+cluster contain up to 2 CPUs running at the same frequency.
+
+Required properties:
+- compatible: must be  "marvell,ap807-cpu-clock";
+- #clock-cells : should be set to 1.
+- clocks : shall be the input parents clock phandle for the clock.
+
+ap_syscon1: system-controller@6f8000 {
+	compatible = "syscon", "simple-mfd";
+	reg = <0x6f8000 0x1000>;
+
+	cpu_clk: clock-cpu {
+		compatible = "marvell,ap807-cpu-clock";
+		clocks = <&ap_clk 0>, <&ap_clk 1>;
+		#clock-cells = <1>;
+	};
+};
-- 
2.17.1

